src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -354,10 +355,21 @@ public JSplitPane(int newOrientation,
                                                "JSplitPane.HORIZONTAL_SPLIT " +
                                                "or JSplitPane.VERTICAL_SPLIT");
         continuousLayout = newContinuousLayout;
-        if (newLeftComponent != null)
-            setLeftComponent(newLeftComponent);
-        if (newRightComponent != null)
-            setRightComponent(newRightComponent);
+        if (this.getComponentOrientation().equals(ComponentOrientation.LEFT_TO_RIGHT)) {
```

Comment:
In the constructor, the orientation cannot be ascertained so the fix is moved to setEnabled